operationsgateway_api/src/records/ingestion/partial_import_checks.py

```diff
@@ -92,15 +100,41 @@ def channel_checks(self):
         accepted_channels = []
         rejected_channels = {}
 
-        for key in list(ingested_channels.keys()):
-            if key in stored_channels:
-                rejected_channels[key] = "Channel is already present in existing record"
+        for channel_name, channel_model in ingested_channels.items():
+            if channel_name in stored_channels:
+                if isinstance(channel_model, (ImageChannelModel, WaveformChannelModel)):
+                    if isinstance(channel_model, ImageChannelModel):
+                        path = Image.get_full_path(channel_model.image_path)
+                    elif isinstance(channel_model, WaveformChannelModel):
+                        path = Waveform.get_full_path(channel_model.waveform_path)
+
+                    object_stored = self._is_image_or_waveform_stored(path)
+
+                if object_stored:
+                    rejected_channels[channel_name] = (
+                        "Channel is already present in existing record"
+                    )
+                else:
+                    accepted_channels.append(channel_name)
```

Comment:
I don't think this will work for non-Image/Waveform channels. `object_stored` is only set on line 111, which can only be reached if the channel model `isinstance(channel_model, (ImageChannelModel, WaveformChannelModel))`. So if that's `False` we're either going to come into this without `object_stored` being set leading to a `NameError` or we use the value of the previous iteration (i.e. the last Image/Waveform we encountered).

Haven't tested this (syntax/functionality might be broken), but I think we'd want something like:

```suggestion
            if channel_name in stored_channels:
                if isinstance(channel_model, ImageChannelModel):
                    path = Image.get_full_path(channel_model.image_path)
                    object_stored = self._is_image_or_waveform_stored(path)
                elif isinstance(channel_model, WaveformChannelModel):
                    path = Waveform.get_full_path(channel_model.waveform_path)
                    object_stored = self._is_image_or_waveform_stored(path)
                else:
                    object_stored = True

                if object_stored:
                    rejected_channels[channel_name] = (
                        "Channel is already present in existing record"
                    )
                else:
                    accepted_channels.append(channel_name)
```

Reply:
Good spot, I've managed to break this when updating a HDF file that contains no image or waveforms channels (an edge case I didn't consider). `UnboundLocalError` was the precise exception I encountered. I'll apply the fix tomorrow.

Reply:
I've just pushed a fix for this - it's pretty much your suggestion but I had to add an additional else statement at the end to fix a bug where accepted channels were no longer appearing in the response.